src/cmd/builtin/python.c

```diff
@@ -16,6 +16,7 @@
 
 static int cmd_python (optparse_t *p, int ac, char *av[])
 {
+    av[0] = PYTHON_INTERPRETER; // Respect symlink'd binaries in virtualenvs
```

Comment:
It should be noted that this affects more than just virtualenv. Before this change, even a non-virtualenv python interpreter will have `av[0]` set to simply `python` (since `av[0]` in a subcommand is always the name of the subcommand). Since python uses `av[0]` to set `sys.executable`, on my system where `PYTHON_INTERPRETER` is `/usr/bin/python3`, `sys.executable` isn't `/usr/bin/python3` or even `python3`, it is just `python`. I guess we've gotten away with it until now.

Maybe we can make the comment here more general, e.g.

```c
/*
 *  Ensure av[0] matches the full path of the interpreter we're executing.
 *  Other than just being good practice, this ensures that sys.executable
 *  is correct (since python uses sys.argv[0] for sys.executable) and so
 *  that symlink'd binaries in virtualenvs are respected.
 */
```